crates/chia_py_streamable_macro/src/lib.rs

```diff
@@ -174,34 +187,60 @@ pub fn py_streamable_macro(input: proc_macro::TokenStream) -> proc_macro::TokenS
     let streamable = quote! {
         #[pyo3::pymethods]
         impl #ident {
-            #[staticmethod]
+            #[classmethod]
             #[pyo3(name = "from_bytes")]
-            pub fn py_from_bytes(blob: pyo3::buffer::PyBuffer<u8>) -> pyo3::PyResult<Self> {
+            pub fn py_from_bytes(cls: &pyo3::Bound<'_, pyo3::types::PyType>, blob: pyo3::buffer::PyBuffer<u8>) -> pyo3::PyResult<pyo3::PyObject> {
+                use pyo3::prelude::PyAnyMethods;
+                use pyo3::IntoPy;
                 if !blob.is_c_contiguous() {
                     panic!("from_bytes() must be called with a contiguous buffer");
                 }
                 let slice = unsafe {
                     std::slice::from_raw_parts(blob.buf_ptr() as *const u8, blob.len_bytes())
                 };
-                <Self as #crate_name::Streamable>::from_bytes(slice).map_err(|e| <#crate_name::chia_error::Error as Into<pyo3::PyErr>>::into(e))
+                let rust_obj = <Self as #crate_name::Streamable>::from_bytes(slice);
```

Comment:
you can use `.map_err()?` here, instead of matching below

Reply:
Done